### DISCUSSION TITLE
Mentions dont show up when I try to use quill.getText()

### DISCUSSION THREAD

**Discussion by alexbennycodes** [afconsult/quill-mention#297] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/98402835/221551908-f62813ec-91e9-4881-8229-bbd2573227d7.png)
adding the mentions do get added to the quill editor
but when I try to display the text using quill.getText(), it doesn't show the mentions
![image](https://user-images.githubusercontent.com/98402835/221552324-bb12533d-0155-4dc7-91b2-85c950583174.png)

**Comment by angelaki** [comment]
Any chance this behavior can be added, soon? I'd help / do it myself, but I'm a complete quill-rookie. So no idea where to start. I guess it would be cool, if a transform-function could be passed (how to translate these mentions to pure text), by default null. So the current behavior won't change / it is not a breaking change.

**Comment by angelaki** [comment, marked as answer]
I just checked the editors `getText()` method and it seams like that's something, _mention_ can't control:

```
  getText(index: number, length: number): string {
    return this.getContents(index, length)
      .filter(op => typeof op.insert === 'string')
      .map(op => op.insert)
      .join('');
  }
```

But since it's quite easy, I've created my own `getText`-method, respecting mentions:

```
quill.getContent().ops
    .filter(op => typeof op.insert === 'string' || op.insert?.mention instanceof DOMStringMap)
    .map(op => typeof op.insert === 'string' ? op.insert : `$$${op.insert.mention.id}`)
    .join('');
```

Even _sexier_, you can just override the prototype:

```
Quill.prototype.getText = function (this: any, index: number, length: number): string {
    return this.getContents(index, length)
        .filter(op => typeof op.insert === 'string' || op.insert?.mention instanceof DOMStringMap)
        .map(op => typeof op.insert === 'string' ? op.insert : `$$${op.insert.mention.id}`)
        .join('');
};
```

Hope this solution is useful for somebody out there.
